server/src/main/java/org/elasticsearch/rest/RestHandler.java

```diff
@@ -86,6 +86,19 @@ default List<Route> routes() {
         return Collections.emptyList();
     }
 
+    /**
+     * The set of path and query parameters that could be present on this handler.
+     * This method is only required due to https://github.com/elastic/elasticsearch/issues/36785,
+     * which conflates query and path parameters inside the rest handler.
+     * This method should be overridden to add path parameters to {@link #supportedQueryParameters}
+     * if the handler has path parameters.
+     * This method will be removed when {@link #supportedQueryParameters()} and {@link BaseRestHandler#responseParams()} are merged.
+     */
+    @Deprecated(forRemoval = true)
```

Comment:
Hmm this puts a red squiggly under overriding implementations of this method as if you're doing something wrong by touching it, but as things stand today it's required. Could we not deprecate it like this until we're ready?